### PR TITLE
Showman Pack

### PR DESCRIPTION
**Pull request by Dayvig**

### CODE REVIEW

File: src/main/java/thePackmaster/powers/showmanpack/HallOfMirrorsPower.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.util.TexLoader;
+
+public class HallOfMirrorsPower extends TwoAmountPower {
+    public static final String POWER_ID = SpireAnniversary5Mod.makeID("HallOfMirrorsPower");
+    private static final PowerStrings powerStrings = CardCrawlGame.languagePack.getPowerStrings(POWER_ID);
+    public static final String NAME = powerStrings.NAME;
+    public static final String[] DESCRIPTIONS = powerStrings.DESCRIPTIONS;
+
+    public HallOfMirrorsPower(AbstractCreature owner, int amount) {
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/showmanpack/HallOfMirrorsPower.java:20]: Please use the packmaster abstract, it already implements two amount power

File: src/main/java/thePackmaster/powers/showmanpack/MagicCylinderPower.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if (info.type != DamageInfo.DamageType.THORNS && info.type != DamageInfo.DamageType.HP_LOSS && info.owner != null && info.owner != this.owner && damageAmount >= owner.currentBlock) {
+            addToBot(new VFXAction(new MagicCylinderEffect(owner.hb.cX + (100f * Settings.scale), owner.hb.cY + (80f * Settings.scale), owner.hb.cX + (100f * Settings.scale), owner.hb.cY + (120f * Settings.scale), 0.5f)));
+            addToBot(new WaitAction(0.5f));
+            addToBot(new VFXAction(new MagicCylinderEffect(owner.hb.cX + (100f * Settings.scale), owner.hb.cY - (80f * Settings.scale), owner.hb.cX + (100f * Settings.scale), owner.hb.cY - (120f * Settings.scale), 0.5f)));
+            addToBot(new WaitAction(0.5f));
+            addToBot(new VFXAction(new MediumLaserBeamEffectReversed(owner.hb.cX + (100f * Settings.scale), owner.hb.cY - (120f * Settings.scale), Color.LIME, 0.4f)));
+            addToBot(new WaitAction(0.5f));
+            addToBot(new DamageAction(info.owner, new DamageInfo(owner, damageAmount, DamageInfo.DamageType.THORNS), AbstractGameAction.AttackEffect.NONE));
+            flash();
+            addToBot(new ReducePowerAction(owner, owner, this, 1));
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/showmanpack/MagicCylinderPower.java:49]: Might be worth to directly reduce the power so you don't trigger the power more than you have chanrges with multi hits

File: src/main/java/thePackmaster/powers/showmanpack/NowYouDontPower.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        }
+        this.description = DESCRIPTIONS[0] + toGenerate.name + DESCRIPTIONS[1];
+    }
+
+    @Override
+    public void atStartOfTurnPostDraw() {
+        if (toGenerate == null){
+            toGenerate = new NowYouDont();
+        }
+        this.addToBot(new MakeTempCardInHandAction(toGenerate.makeCopy()));
```
Comment (erasels) [inline src/main/java/thePackmaster/powers/showmanpack/NowYouDontPower.java:38]:
```suggestion
        this.addToBot(new MakeTempCardInHandAction(toGenerate));
```
That action already makes a copy

File: src/main/resources/anniv5Resources/localization/eng/showmanpack/Cardstrings.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{
+  "${ModID}:Dazzle": {
+    "NAME": "Dazzle",
+    "DESCRIPTION": "Deal !D! damage. NL Apply !M! Weak."
+  },"${ModID}:Amaze": {
+  "NAME": "Amaze",
+  "DESCRIPTION": "ALL enemies lose !M! HP for each stack of Weak they have."
+},"${ModID}:Grandstand": {
+  "NAME": "Grandstand",
+  "DESCRIPTION": "Gain !B! Block. NL Next turn, gain [E] .",
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/showmanpack/Cardstrings.json:10]: Look into dynamic descriptions for this, would be nice.

**Review by erasels — CHANGES_REQUESTED** [review]
Ignoring the points from the other PR just makes me bring them back up

**Comment by erasels** [comment]
@Dayvig any progress on this or did you lose interest?

**Comment by Dayvig** [comment]
Just haven't gotten to it yet. I have some time today so I'll take another
look at the lingering issues-

**Pull request merged**